hack/getting_started.sh

```diff
@@ -0,0 +1,592 @@
+#!/bin/bash
+function isCmdInstalled() {
+    local cmd=$1
+    ${cmd} version 2>&1 | grep -q "not found"
```

Comment:
I tried to remove version, but found issue with "jq 2>&1 | grep -q "not found"", it hangs.
Now all commands I used are work for "** version" right now. If there is command not work for version in the future, I'll do something in this function.

Reply:
@huchen2021 @anusha94 what we observed here https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/blob/main/agent/installer/checks.go#L14-L30 -> Potential of reuse if we extract a go lib out of it?

Reply:
In the meantime check if "command -v" is a better alternative within this script base checker for a binary/command?
https://pubs.opengroup.org/onlinepubs/9699919799/utilities/command.html
```
$ command -V docker clusterctl
docker is /usr/local/bin/docker
clusterctl not found
```
or

```
$ command -v docker clusterctl
/usr/local/bin/docker
```

Reply:
> @huchen2021 @anusha94 what we observed here https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/blob/main/agent/installer/checks.go#L14-L30 -> Potential of reuse if we extract a go lib out of it?

How to reuse go function on shell script?

Reply:
> In the meantime check if "command -v" is a better alternative within this script base checker for a binary/command? https://pubs.opengroup.org/onlinepubs/9699919799/utilities/command.html
> 
> ```
> $ command -V docker clusterctl
> docker is /usr/local/bin/docker
> clusterctl not found
> ```
> 
> or
> 
> ```
> $ command -v docker clusterctl
> /usr/local/bin/docker
> ```

command is a bash builtin which is not installed by default. It is not installed on customer's vm, it may needs to install it by "sudo apt-get install manpages-posix".  I am leaning towards not to have this one more dependency.

Reply:
Suggestion: if we only wanted to see the tool is installed, Is `which` utility an option?

Reply:
@dharmjit Sounds a good idea.